http-clients/src/main/java/org/triplea/http/client/moderator/toolbox/ModeratorToolboxFeignClient.java

```diff
@@ -0,0 +1,53 @@
+package org.triplea.http.client.moderator.toolbox;
+
+import java.util.List;
+import java.util.Map;
+
+import org.triplea.http.client.HttpConstants;
+
+import feign.HeaderMap;
+import feign.Headers;
+import feign.Param;
+import feign.RequestLine;
+
+/**
+ * Http client for moderator 'toolbox' actions. The toolbox is essentially a set of windows
+ * that show lobby table data and provide CRUD operations to moderators. For example
+ * with the toolbox moderators should be able to update the list of restricted usernames,
+ * view players that have joined the lobby, add and remove player bans, etc.
+ * <p>
+ * This set of http methods should all require an API key passed via headers to authenticate
+ * the moderator issuing the request.
+ * </p>
+ * <p>
+ * Rate-limiting: of note, the backend implementation should be careful to apply rate limiting
+ * to any/all endpoints that take an API key so as to avoid brute-force attacks to try and crack
+ * an API key value.
+ * </p>
+ */
+interface ModeratorToolboxFeignClient {
```

Comment:
Of some note, this class is package-protected and provides a "Raw" API. To avoid any user of this interface from knowing how to create headers, we have a public "ModeratorToolboxClient" which takes care of those details. So for example instead of providing a map parameter, we can request a String parameter to pass the api key that is then passed to this class as a proper map.